chatkit-core/src/test/kotlin/com/pusher/chatkit/rooms/state/JoinedRoomsReceivedIntegrationTest.kt

```diff
@@ -0,0 +1,234 @@
+import com.pusher.chatkit.users.api.UserSubscriptionEvent
+import com.pusher.chatkit.util.DateApiTypeMapper
+import io.mockk.confirmVerified
+import io.mockk.mockk
+import io.mockk.verify
+import org.reduxkotlin.Dispatcher
+import org.spekframework.spek2.Spek
+import org.spekframework.spek2.style.specification.describe
+
+internal object JoinedRoomsReceivedIntegrationTest : Spek({
+
+    fun generateApiTypeRoom(id: String, name: String) = JoinedRoomApiType(
+        id = id,
+        name = name,
+        createdById = "person1",
+        pushNotificationTitleOverride = "notification override",
+        private = false,
+        customData = mapOf("highlight" to "blue"),
+        lastMessageAt = "2020-02-27T17:12:10Z",
+        updatedAt = "2020-02-27T17:12:20Z",
+        createdAt = "2020-02-27T17:12:30Z",
+        deletedAt = null
+    )
+    val roomApiTypeOne = generateApiTypeRoom("id1", "room1")
+    val roomApiTypeOneUpdated = generateApiTypeRoom("id1", "room1-updated")
+    val roomApiTypeTwo = generateApiTypeRoom("id2", "room2")
+    val roomApiTypeThree = generateApiTypeRoom("id3", "room3")
+
+    val simpleUser = UserApiType(
+        id = "user1",
+        createdAt = "",
+        updatedAt = "",
+        name = "name",
+        avatarURL = null,
+        customData = null,
+        online = false
+    )
+
+    describe("given existing state with one room") {
+        val dateApiTypeMapper = DateApiTypeMapper()
+        val joinedRoomApiTypeMapper = JoinedRoomApiTypeMapper(dateApiTypeMapper)
+        val dispatcher by memoized { mockk<Dispatcher>(relaxed = true) }
+        val testState = State(joinedRoomsState = JoinedRoomsState(
+            rooms = mapOf(
+                "id1" to joinedRoomApiTypeMapper.toRoomInternalType(roomApiTypeOne)
+            ),
+            unreadCounts = mapOf(
+                "id1" to 1))
+        )
+        val differ = JoinedRoomsStateDiffer { testState }
+        val userSubscriptionDispatcher by memoized {
+            UserSubscriptionDispatcher(
+                joinedRoomApiTypeMapper = joinedRoomApiTypeMapper,
+                joinedRoomsStateDiffer = differ,
+                dispatcher = dispatcher
+            )
+        }
+
+        describe("when the same InitialState event is received") {
+            val event = UserSubscriptionEvent.InitialState(
+                currentUser = simpleUser,
+                rooms = listOf(roomApiTypeOne),
+                readStates = listOf(
+                    RoomReadStateApiType("id1", 1, null)
+                ),
+                memberships = listOf(
+                    RoomMembershipApiType("id1", listOf())
+                )
+            )
+
+            beforeEachTest {
+                userSubscriptionDispatcher.onEvent(event)
+            }
+
+            it("then no actions are dispatched") {
+                verify(exactly = 0) {
+                    dispatcher(any())
+                }
+                confirmVerified(dispatcher)
```

Comment:
minor: redundant as dispatcher cannot be called with anything else (there's nothing more generic than `dispatcher(any())` for this `fun`)